src/Database/LSMTree/Common.hs

```diff
@@ -255,3 +251,9 @@ mkSnapshotName s
   where
     len = length s
     isValid c = ('a' <= c && c <= 'z') || ('0' <= c && c <= '9' ) || c `elem` "-_"
+
+{-------------------------------------------------------------------------------
+  Snapshot name
+-------------------------------------------------------------------------------}
+
+data BlobRef m blob = forall h. Eq h => BlobRef (Internal.BlobRef m (Internal.Run h) blob)
```

Comment:
Ok, and similarly, the internal type exposes the `h` param but we're hiding it in the exported API.